From: Dev Jain <dev.jain@arm.com>
Subject: maple tree: use goto label to simplify code
Date: Tue, 24 Jun 2025 13:37:48 +0530

Use the underflow goto label to set the status to ma_underflow and return
NULL, as is being done elsewhere.

Link: https://lkml.kernel.org/r/20250624080748.4855-1-dev.jain@arm.com
Signed-off-by: Dev Jain <dev.jain@arm.com>
Reviewed-by: Liam R. Howlett <Liam.Howlett@oracle.com>
Cc: Wei Yang <richard.weiyang@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/maple_tree.c |    7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

--- a/lib/maple_tree.c~maple-tree-use-goto-label-to-simplify-code
+++ a/lib/maple_tree.c
@@ -4565,11 +4565,8 @@ again:
 		return entry;
 
 	if (!empty) {
-		if (mas->index <= min) {
-			mas->status = ma_underflow;
-			return NULL;
-		}
-
+		if (mas->index <= min)
+			goto underflow;
 		goto again;
 	}
 
_
